desc/integrals/singularities.py

```diff
@@ -433,91 +554,110 @@ def eval_pt_loop(i, fj):
-    v = eta * s**2 * h_t * h_z / 4 * abs(r) * dr * dw
-    keys = list(set(["|e_theta x e_zeta|"] + kernel.keys))
+    v = (
+        _chi(r)
+        * (interpolator.ht * interpolator.hz)
+        * (interpolator.st * interpolator.sz / 4)
+        * r
+        * dr
+        * dw
+    )
+
+    keys = set(["|e_theta x e_zeta|"] + kernel.keys)
     if "phi" in keys:
-        keys += ["omega"]
+        keys.remove("phi")
+        keys.add("omega")
+    keys = list(keys)
     fsource = [source_data[key] for key in keys]
+    # Note that it is necessary to take the Fourier transforms of the
+    # vector coordinates of the orthonormal polar basis vectors R̂, ϕ̂, Ẑ.
+    # Vector coordinates in the Cartesian basis are not NFP periodic.
+    if isinstance(interpolator, DFTInterpolator):
+        fsource = [interpolator.fourier(val) for val in fsource]
+        is_fourier = True
+    else:
+        # TODO: https://github.com/f0uriest/interpax/issues/53.
+        is_fourier = False
 
-    def polar_pt_vmap(i):
-        # evaluate the effect from a single polar node around each eval point
-        # on that eval point. Polar grids from other singularities have no effect
-        # See sec 3.2.2 of [2]
-        dt = s / 2 * h_t * r[i] * jnp.sin(w[i])
-        dz = s / 2 * h_z * r[i] * jnp.cos(w[i])
-        theta_i = eval_theta + dt
-        zeta_i = eval_zeta + dz
+    def polar_pt(i):
+        """See sec 3.2.2 of [2].
 
-        # data interpolated to each eval pt offset by dt,dz
+        Evaluate the effect from a single polar node around each eval point
+        on that eval point. Polar grids from other singularities have no effect.
+        """
+        # The ``FFTInterpolator`` interpolates to polar node i of the source grid, while
+        # the ``DFTInterpolator`` interpolates to polar node i of the eval   grid.
+        # When the source grid and eval grid have different NFP, the values taken
+        # by a function that is periodic with source grid NFP will in general
+        # be different at these points. For functions with no toroidal variation
+        # there will be no difference, and that is the only time the
+        # source grid and eval grid may have different NFP.
+        vander = interpolator.vander_polar(i)
         source_data_polar = {
-            key: interpolator(val, i) for key, val in zip(keys, fsource)
+            key: interpolator(val, i, is_fourier=is_fourier, vander=vander)
+            for key, val in zip(keys, fsource)
         }
-
-        # can't interpolate phi directly since its not periodic, so we interpolate
-        # omega and add it back in
-        source_data_polar["zeta"] = zeta_i
-        source_data_polar["theta"] = theta_i
-        if "phi" in keys:
-            source_data_polar["phi"] = zeta_i + source_data_polar["omega"]
+        # The (θ, ζ) coordinates at which the maps above were evaluated.
+        # For FFT interpolator on functions with toroidal variation used on
```

Comment:
Commit this before approving the PR
```suggestion
        # For FFT interpolator on functions with no toroidal variation used on
```